src/status_im/wallet/core.cljs

```diff
@@ -610,14 +608,11 @@
             {:db (assoc-in db [:wallet/prepare-transaction field] value)}
             (bottom-sheet/hide-bottom-sheet-old)))
 
-(rf/defn navigate-to-recipient-code
-  {:events [:wallet.send/navigate-to-recipient-code]}
-  [{:keys [db] :as cofx}]
-  (rf/merge cofx
-            {:db (-> db
-                     (assoc :wallet/recipient {}))}
-            (bottom-sheet/hide-bottom-sheet-old)
-            (navigation/open-modal :recipient nil)))
+(re-frame/reg-event-fx :wallet.send/navigate-to-recipient-code
+ (fn [{:keys [db]}]
+   {:db (-> db (assoc :wallet/recipient {}))
+    :fx [[:dispatch [:bottom-sheet/hide-old]]
```

Comment:
FYI: Like me, you may appreciate how symmetrical the code looks because `db` and `fx` have two characters and are the only recommended built-in effects to be returned.

Reply:
this is a big improvement imo as it appears in the old method using `rf/merge` we are mixing two variants of the same api. 
e.g  the event defn name `(bottom-sheet/hide-bottom-sheet-old)` and event dispatch names. 

Additionally it can be quite confusing (as seen in this example) the dispatch [:bottom-sheet/hide-old] has a different name to it's corresponding defn name `(bottom-sheet/hide-bottom-sheet-old)`